### PR TITLE
feat: expose create/list/delete volumes for extensions

### LINKED ISSUE #5564: containerEngine in extensionApi should support volumes operation
### Is your enhancement related to a problem? Please describe

containerEngine already support 
- listContainers
- listImages
- listNetworks
etc.

### Describe the solution you'd like

Adding
- listVolumes
- createVolume
- deleteVolume

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Add new operations for handling volumes from extensions

### What issues does this PR fix or reference?

fixes https://github.com/containers/podman-desktop/issues/5564

### How to test this PR?

unit tests provided

### CODE REVIEW

**Comment by benoitf** [comment]
rebased

**Review by cdrage — APPROVED** [review]

**Review by deboer-tim — APPROVED** [review]

File: packages/extension-api/src/extension-api.d.ts [hunk 1]
```diff
@@ -2101,6 +2101,51 @@ declare module '@podman-desktop/api' {
...
+    engineName: string;
+    CreatedAt: string;
+    containersUsage: { id: string; names: string[] }[];
+    Name: string;
+    Driver: string;
+    Mountpoint: string;
+    Status?: { [key: string]: string };
+    Labels: { [key: string]: string };
+    Scope: 'local' | 'global';
+    Options?: { [key: string]: string } | null;
```
Comment (axel7083) [inline packages/extension-api/src/extension-api.d.ts:2115]: The usage of null is not supposed to be prohibited by the linter ? Is this definition from the kube api itself ?

Comment (benoitf) [reply packages/extension-api/src/extension-api.d.ts:2115]: definition comes from 3rd party libs (where null is not prohibited)

Comment (benoitf) [reply packages/extension-api/src/extension-api.d.ts:2115]: so if we return an object not matching the definition, there is a conflict

**Review by axel7083 — APPROVED** [review]
LGTM

**Pull request merged**